plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/kotlin/KSTypeImpl.kt

```diff
@@ -49,7 +49,7 @@ class KSTypeImpl private constructor(
         ksTypeArguments ?: kotlinType.arguments.map { KSTypeArgumentDescriptorImpl.getCached(it) }
     }
 
-    override fun isAssignableFrom(that: KSType): Boolean = (that as KSTypeImpl).kotlinType.isSubtypeOf(kotlinType)
+    override fun isAssignableFrom(that: KSType): Boolean = (that as? KSTypeImpl)?.kotlinType?.isSubtypeOf(kotlinType) == true
```

Comment:
Would be better to throw an exception when `that` is not applicable:
```
when (that) {
  is KSTypeImpl -> that.kotlinType.isSubtypeOf(kotlinType)
  is KSErrorType -> false
  else -> throw ...
}
```

Reply:
Makes no difference for now since these are the only 2 implementations of KSType, but I can do that as well.

Reply:
There is a different for non-KSType.